### ISSUE TITLE
Stillhalter-Kostenbasis: Per-Lot Strike-Zuordnung statt FIFO-Verteilung

### ISSUE THREAD

**Issue by KonvexInvestment** [KonvexInvestment/ibkr-steuer#32]
## Problem

Bei Put-Andienungen mit **mehreren Strikes auf dem gleichen Underlying** weicht die korrigierte Aktien-Kostenbasis vom jeweiligen Strike-Preis ab.

### Ursache

Die Prämien-Korrekturen (`pending_stk_corrections`) werden per **Underlying-Symbol** gesammelt und dann FIFO auf alle Aktien-Verkäufe dieses Symbols verteilt. Die Zuordnung "welcher Aktien-Verkauf kam aus welcher Put-Andienung (= welchem Strike)" geht verloren.

### Beispiel: RSP (aus echten Audit-Daten)

Zwei Put-Serien auf RSP:
- **Put $156** → 2 Kontrakte assigned → 200 Aktien, Kostenbasis soll $156/Stk
- **Put $157** → 4 Kontrakte assigned → 400 Aktien, Kostenbasis soll $157/Stk

Aktien-Verkäufe:
- Verkauf 1: 100 Stk RSP → Kostenbasis/Stk = **$160.36** (soll: $156.00)
- Verkauf 2: 500 Stk RSP → Kostenbasis/Stk = **$156.76** (soll: Mix aus $156/$157)

Die FIFO-Verteilung der Prämien-Korrekturen matcht nicht die tatsächliche Strike-Zuordnung. Bei nur einem Strike (z.B. SVOL $23) stimmt die Kostenbasis exakt.

### Beispiel: TLT (aus echten Audit-Daten)

Noch ausgeprägter — viele verschiedene Strikes auf dem gleichen Underlying:
- Put $100, Put $96.5, Put $96, Put $95.5, Put $93, Put $95 etc.
- Jeweils separate Andienungen → Aktien mit verschiedenen Strike-Kostenbasis
- FIFO-Verteilung mischt die Korrekturen über alle Strikes

Auswirkung bei TLT:
| Aktien-Verkauf | Kostenbasis/Stk | Zugehöriger Strike | Abweichung |
|---|---|---|---|
| 100 Stk 01.01 | $100.64 | $100 | +$0.64 |
| 200 Stk 09.01 | $100.07 | $96 | +$4.07 |
| 200 Stk 16.02 | $96.10 | $93 | +$3.10 |

## Vorgeschlagene Lösung

Statt `pending_stk_corrections` nur per Underlying-Symbol zu indizieren, die Zuordnung per **Assignment-Datum** oder **CLOSED_LOT `openDateTime`** herstellen:

1. **Option A: CLOSED_LOT Matching** — Jedes CLOSED_LOT hat ein `openDateTime` (= Andienungsdatum). Über dieses Datum lässt sich der zugehörige Put-Strike identifizieren, und die Prämien-Korrektur pro Lot exakt zuordnen.

2. **Option B: Assignment-Datum Matching** — Das Andienungsdatum der Aktie (= BookTrade-Datum des Puts) wird als Schlüssel verwendet, um `pending_stk_corrections` nicht nur per Symbol, sondern per `(symbol, assignment_date)` zu indizieren. Aktien-Trades mit `openDateTime` am Assignment-Datum bekommen die richtige Korrektur.

Option A ist präziser (per-Lot statt per-Trade), Option B einfacher.

## Abgrenzung

Dieser Issue ist **unabhängig** vom Fix der zurückgekauften Sells in der Prämienberechnung (`_get_open_option_sells`). Jener Bug ist bereits gefixt. Das hier ist eine Verbesserung der Zuordnungsgenauigkeit bei Multi-Strike-Underlyings.

## Betroffene Stellen

- `calculate_tax_report.py` ~Zeile 900-933: `pending_stk_corrections` Aufbau und Anwendung
- `calculate_tax_report.py` ~Zeile 1486-1528: Cross-Year Put-Korrekturen (gleiche Logik)

**Comment by KonvexInvestment** [comment]
### Zusätzlicher Impact: Gain/Loss-Split bei Stillhalter-Korrekturen

Beim Fix für negative `stocks_gain` (pauschal `stocks_gain -= stk_premium` → per-Trade Gain/Loss-Split) hat sich gezeigt, dass die FIFO-Zuordnung aus #32 auch den **Topf-Split** beeinflusst.

Die Korrektur wird aktuell dem ersten passenden Trade per Symbol zugeordnet, nicht dem Trade, in dem IBKR die Prämie tatsächlich eingebettet hat. Wenn die Trades unterschiedliche Vorzeichen haben, weicht der Split ab:

```
Verkauf 1: pnl = +25 EUR (Gewinn)    ← wird korrigiert (falsch zugeordnet)
Verkauf 2: pnl = -10 EUR (Verlust)   ← hier steckt die Prämie eigentlich

Korrektur = 55 EUR
Aktuell:  from_gain = 25, from_loss = 30
Korrekt:  from_gain = 0,  from_loss = 55
```

Per-Lot Matching (Option A aus dem Issue) würde beides lösen: korrekte Kostenbasis im Excel UND korrekter Gain/Loss-Split.
